src/components/vm/nics/vmNicsCard.tsx

```diff
@@ -440,49 +453,71 @@ export class VmNetworkTab extends React.Component<VmNetworkTabProps, VmNetworkTa
             {
                 name: _("IP address"),
                 value: (network, networkId) => {
-                    const ips = this.state.ips.filter(ip => ip.mac === network.mac);
-
-                    const ip: { inet?: string, inet6?: string } = {};
-                    // If information from agent are available, it's preferred over lease/arp
-                    ips.sort((a) => a.source === "agent" ? -1 : 1).forEach(a => {
-                        // Various sources (agent,lease,arp) may have only partial information about interface
-                        // (e.g. one source may only know ipv4 address, the other may only know ipv6 address)
-                        // Therefore, look at each source to get all available information
-                        if (!ip.inet && a.ip.inet)
-                            ip.inet = a.ip.inet;
-                        if (!ip.inet6 && a.ip.inet6)
-                            ip.inet6 = a.ip.inet6;
+                    const ipInterfaces = this.state.ips.filter(ip => ip.mac === network.mac);
+
+                    const inetIps: string[] = [];
+                    const inet6Ips: string[] = [];
+
+                    // Setup a preferred order of IP/SUBNET-MASK we want to add.
+                    // Helps ensure that the IP subnet mask are taken from the right source.
+                    // agent > lease > arp
+                    ipInterfaces.sort((a, b) => {
+                        if (a.source === "agent")
+                            return -1;
+                        else if (a.source === "lease" && b.source !== "agent")
+                            return -1;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2303-0f9a39db-20250905-160146-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)